src/swarm/util/RecordBatcher.d

```diff
@@ -647,3 +647,31 @@ unittest
     test!("==")(i, 2);
 }
 
+// Test extracting a larger batch than the reader was initialised for. (This is
+// silently allowed.)
+unittest
+{
+    auto lzo = new Lzo;
+    auto writer = new RecordBatcher(lzo, RecordBatcher.DefaultMaxBatchSize * 10);
+    auto reader = new RecordBatch(lzo, RecordBatcher.DefaultMaxBatchSize);
+
+    ubyte[] compressed;
+    uint added;
+    while ( true )
+    {
+        auto res = writer.add(format("0x{:16}", added),
+            format("{}{}{}", added, added, added));
+        if ( res == res.BatchFull )
+            break;
+        added++;
+    }
+
+    writer.compress(compressed);
+    assert(compressed.length > RecordBatcher.DefaultMaxBatchSize);
```

Comment:
This checks the length of the compressed data, but wasn't the restriction on the uncompressed batch? (If the compressed data are longer then the uncompressed will usually be longer, too, just checking if there is a misconception.)

Reply:
This line confirms that the compressed batch is too big to fit in the default initialised size of the batch reader, which is the whole point of this test. If the condition is not true, the test is meaningless.